src/rankstr_mpi.c

```diff
@@ -586,9 +586,11 @@ void rankstr_mpi(const char* str, MPI_Comm comm, int tag1, int tag2, int* groups
   //int tag1 = 0;
   //int tag2 = 1;
 
-  /* require str not be NULL */
+  /* rankstr requires a valid string,
+   * so use the empty string for NULL strings */
+  const char emptystr[] = "";
   if (str == NULL) {
-    /* TODO: error */
+    str = emptystr;
   }
```

Comment:
This change makes NULL strings and empty strings equivalent.  Let's leave this so we distinguish between these two cases.  This could be useful in some applications, and it is consistent with how ``rankstr_mpi_comm_split`` behaves.  In ``rankstr_mpi_comm_split``, processes that specify an empty string end up in a communicator with any other procs that gave an empty string, while procs that pass in NULL end up with ``MPI_COMM_NULL``.

Long term it might make sense to set ``groupid = MPI_UNDEFINED``.  The problem is that the code in the remainder of the function does not properly handle an input of a NULL string.  We should leave this as an error condition, and for now, we can call MPI_Abort here.